library/ssl_tls12_server.c

```diff
@@ -962,7 +973,18 @@ static int ssl_pick_cert( mbedtls_ssl_context *ssl,
         MBEDTLS_SSL_DEBUG_CRT( 3, "candidate certificate chain, certificate",
                           cur->cert );
 
+#if defined(MBEDTLS_USE_PSA_CRYPTO)
+#if defined(MBEDTLS_SSL_ASYNC_PRIVATE)
+        if( ( ssl->conf->f_async_sign_start == NULL &&
```

Comment:
Could you elaborate on this set of conditions? Why are we ignoring `cur->cert->pk` if `MBEDTLS_SSL_ASYNC_PRIVATE` is not defined, and why do we check `cur->key` only if sign, dec functions are not present?

Reply:
If async sign or decrypt is enabled, the cert pk can be loaded later (asynchronously) so we check the cert key instead because the cert pk is probably invalid in this case.